ivy/array/general.py

```diff
@@ -1,8 +1,64 @@
 # global
 import abc
-
 # ToDo: implement all methods here as public instance methods
 
+# local
+import ivy
+
 
 class ArrayWithGeneral(abc.ABC):
-    pass
+    def all_equal(
+            self,
+            x,
+            equality_matrix: bool = False
+    ):
+        """Determines whether the inputs are all equal.
+
+        Parameters
+        ----------
+        xs
+            inputs to compare.
+        equality_matrix
+            Whether to return a matrix of equalities comparing each
+            input with every other.
+            Default is False.
+
+        Returns
+        -------
+        ret
+            Boolean, whether or not the inputs are equal, or matrix array of booleans
+            if equality_matrix=True is set.
+
+        Instance Method Examples
```

Comment:
Both examples seem to be doing the same thing, please remove one example from each so that we have one example for `With one ivy.Array instance method` and one example for `With a mix of ivy.Array and ivy.NativeArray instance method`

Reply:
Also, the header can simply be `Examples`